### ISSUE TITLE
Face similarity with opencv

### ISSUE THREAD

**Issue by PanMig** [ox-vgg/vgg_face2#45]
I would like to use the vggFace2 for computing the face similarity between two input images.

I am using the OpenCV inference engine so no training will take place but only feature extraction. Based on the paper, (using the caffe resnet-50 model ) I need the features (2048) from the layer adjacent in the classifier layer. In my case, when iterating through the layer names I see that this layer is named "pool5/7x7_s1", is this correct?

Moreover, after we extract the features do we need to L2 normalize the vector? Last but not least can also Euclidean distance be used for similarity or only cosine similarity?

**Comment by WeidiXie** [comment]
HI, could you read this please ?
https://github.com/ox-vgg/vgg_face2/blob/master/standard_evaluation/pytorch_feature_extractor.py
